src/cpp/rtps/builtin/discovery/participant/PDPClient.cpp

```diff
@@ -933,412 +933,6 @@ void PDPClient::match_pdp_reader_nts_(
-
-                                    if (IPLocator::isAny(server_locator))
-                                    {
-                                        // A server cannot be reach in all interfaces, it's clearly a localhost call
-                                        IPLocator::setIPv4(server_locator, "127.0.0.1");
-                                    }
-
-                                    process_port(port, server_locator);
-                                    IPLocator::setLogicalPort(server_locator, static_cast<uint16_t>(port));
-                                    flist.push_front(server_locator);
-                                }
-
-                                for ( const std::string& loc : ipv6 )
-                                {
-                                    server_locator.kind = LOCATOR_KIND_TCPv6;
-                                    server_locator.set_Invalid_Address();
-                                    IPLocator::setIPv6(server_locator, loc);
-
-                                    if (IPLocator::isAny(server_locator))
-                                    {
-                                        // A server cannot be reach in all interfaces, it's clearly a localhost call
-                                        IPLocator::setIPv6(server_locator, "::1");
-                                    }
-
-                                    process_port(port, server_locator);
-                                    IPLocator::setLogicalPort(server_locator, static_cast<uint16_t>(port));
-                                    flist.push_front(server_locator);
-                                }
-                            }
-                        }
-                    }
-
-                    if (flist.empty())
-                    {
-                        std::stringstream ss;
-                        ss << "Wrong domain name passed into the server's list " << locator;
-                        throw std::invalid_argument(ss.str());
-                    }
-
-                    // Add server to the list
-                    add_server2qos(std::move(flist), servers_list);
-                }
-                else
-                {
-                    std::stringstream ss;
-                    ss << "Wrong locator passed into the server's list " << locator;
-                    throw std::invalid_argument(ss.str());
-                }
-            }
-
-            // Advance to the next server if any
-            ++server_it;
-        }
-
-        // Check for server info
-        if (servers_list.empty())
-        {
-            throw std::invalid_argument("No default server locators were provided.");
-        }
-    }
-    catch (std::exception& e)
-    {
-        EPROSIMA_LOG_ERROR(SERVER_CLIENT_DISCOVERY, e.what());
-        servers_list.clear();
-        return false;
-    }
-
-    return true;
-}
-
 GUID_t RemoteServerAttributes::GetParticipant() const
```

Comment:
Implementation of `RemoteServerAttributes` methods shall be moved to `ServerAttributes.cpp`